tasks/main.yml

```
@@ -1,2 +1,65 @@
 ---
-# tasks file for skeleton
+- name: Load var file with default values based on the OS type
+  ansible.builtin.include_vars: "{{ lookup('first_found', params) }}"
+  vars:
+    params:
+      files:
+        - >-
+          {{ ansible_distribution }}_{{ ansible_distribution_release }}_{{
+          ansible_architecture }}.yml
+        - "{{ ansible_distribution }}_{{ ansible_distribution_release }}.yml"
+        - "{{ ansible_distribution }}_{{ ansible_architecture }}.yml"
+        - "{{ ansible_distribution }}.yml"
+        - "{{ ansible_os_family }}_{{ ansible_architecture }}.yml"
+        - "{{ ansible_os_family }}.yml"
+        - Default_{{ ansible_architecture }}.yml
+        - Default.yml
+      paths:
+        - "{{ role_path }}/vars"
+
+- name: Add archive repositories
+  when: ansible_distribution_release in supported_releases | default([])
+  block:
+    - name: Remove non-archive repositories
+      ansible.builtin.file:
+        path: /etc/apt/sources.list
```

I think you meant to do this?

```suggestion
        path: "{{ item }}"
```